src/plotting/plotSettings.component.tsx

```diff
@@ -80,6 +81,94 @@ const StyledTab = styled(Tab)(() => ({
   width: 10,
 }));
 
+/**
+ * Handles the colours currently in use for plotting different channels
+ * Determines which colours we have available based on which have already been selected
+ * Uses a list of 10 pre-selected colours before generating a random colour beyond this
+ */
+export class ColourGenerator {
```

Comment:
Wow - I was just going to iterate over the array of 10-ish colours and not care about reusing colours, but this is super cool! Nice solution!